src/PrestaShopBundle/Controller/Admin/Improve/Design/MailThemeController.php

```diff
@@ -0,0 +1,364 @@
+                            'Admin.Notifications.Success',
+                            [
+                                $data['mail_theme'],
+                                $data['language'],
+                            ]
+                        )
+                    );
+                } else {
+                    $this->addFlash(
+                        'success',
+                        $this->trans(
+                            'Successfully generated mail templates for theme %s with locale %s',
+                            'Admin.Notifications.Success',
+                            [
+                                $data['mail_theme'],
+                                $data['language'],
+                            ]
+                        )
+                    );
+                }
+            } catch (CoreException $e) {
+                $this->flashErrors([
+                    $this->trans(
+                        sprintf(
+                            'Could not generate mail templates for theme %s with locale %s',
+                            $data['mail_theme'],
+                            $data['language']
+                        ),
+                        'Admin.Notifications.Error'
+                    ),
+                    $e->getMessage(),
+                ]);
+            }
+        }
+
+        return $this->redirectToRoute('admin_mail_theme_index');
+    }
+
+    /**
+     * Save mail theme configuration
+     *
+     * @AdminSecurity("is_granted(['update'], request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     *
+     * @return Response
+     *
+     * @throws \Exception
+     */
+    public function saveConfigurationAction(Request $request)
+    {
+        /** @var FormHandlerInterface $formHandler */
+        $formHandler = $this->getMailThemeFormHandler();
+        /** @var Form $form */
+        $form = $formHandler->getForm()->handleRequest($request);
+
+        if ($form->isSubmitted()) {
+            if (!$form->isValid()) {
+                $this->flashErrors($this->getFormErrorsForJS($form));
+
+                return $this->redirectToRoute('admin_mail_theme_index');
+            }
+
+            $errors = $formHandler->save($form->getData());
+            if (empty($errors)) {
+                $this->addFlash(
+                    'success',
+                    $this->trans(
+                        'Mail theme configuration saved successfully',
+                        'Admin.Notifications.Success'
+                    )
+                );
+            } else {
+                $this->flashErrors($errors);
+            }
+        }
+
+        return $this->redirectToRoute('admin_mail_theme_index');
+    }
+
+    /**
+     * @param Request $request
+     * @param string $themeName
+     *
+     * @return Response
+     *
+     * @throws InvalidArgumentException
+     */
+    public function previewThemeAction(Request $request, $themeName)
```

Comment:
We don't need any rights to execute this action?

Reply:
We do! My bad I forgot it, thanks